src/cms.c

```diff
@@ -0,0 +1,96 @@
+#include <stdlib.h>         // malloc
+#include <string.h>         // strlen
+#include <stdio.h>          // printf
+#include <math.h>           // q, ceil
+#include <assert.h>         // assert
+
+#include "cms.h"
+#include "contrib/xxhash.h"
+
+
+CMSketch *NewCMSketch(size_t width, size_t depth) {
+    assert(width > 0);
+    assert(depth > 0);
+
+    CMSketch *cms = CMS_CALLOC(1, sizeof(CMSketch));
+
+    cms->width = width;
+    cms->depth = depth;
+    cms->counter = 0;
+    cms->array = CMS_CALLOC(width * depth, sizeof(size_t));
+
+    return cms; 
+}
+
+void CMS_Destroy(CMSketch *cms) {
+    assert(cms);
+
+    CMS_FREE(cms->array);
+    cms->array = NULL;
+
+    CMS_FREE(cms);
+}
+
+void CMS_IncrBy(CMSketch *cms, const char *item, size_t value) {
+    assert(cms);
+    assert(item);
+
+    for(size_t i = 0; i < cms->depth; ++i) {
+        size_t hash = XXH64(item, strlen(item), i);
+        cms->array[(hash % cms->width) + (i * cms->width)] += value;
+    }
+    cms->counter += value;
+}
+
+size_t CMS_Query(CMSketch *cms, const char *item) {
+    assert(cms);
+    assert(item >= 0);
+
+    size_t temp = 0, res = ~0;
```

Comment:
`~0LLU`, `-1`, `SIZE_MAX`, or somesuch. I'm not 100% sure that `~0` does what you think it does. I've had odd moments where it didn't set all the bits (:o)